src/pages/api/get-open-api-docs-view-props.ts

```diff
@@ -0,0 +1,113 @@
+import { getStaticProps } from 'views/open-api-docs-view/server'
+// Types
+import type { NextApiRequest, NextApiResponse } from 'next'
+import type { OpenAPIV3 } from 'openapi-types'
+import type { ProductSlug } from 'types/products'
+
+/**
+ * Boilerplate page configuration, we could in theory expose this so visitors
+ * to the preview tool could manipulate it, but we intentionally just
+ * hard-code here to keep the focus of the preview tool on OpenAPI spec
+ * contents.
+ */
+const GENERIC_PAGE_CONFIG = {
+	// basePath same no matter what, preview tool is on static route
+	basePath: '/openapi-docs-preview',
+	// No versioning in the preview tool, focus on one spec file at a time
+	context: { params: { page: [] } },
+	// Product slug, using HCP to just show a generic HashiCorp logo,
+	// so that the preview tool's focus can remain on the spec file contents
+	productSlug: 'hcp' as ProductSlug,
+	// Generic resource items, we can set more specific ones closer to launch
+	navResourceItems: [
+		{
+			title: 'Tutorial Library',
+			href: '/tutorials/library',
+		},
+		{
+			title: 'Certifications',
+			href: '/certifications',
+		},
+		{
+			title: 'Community',
+			href: 'https://discuss.hashicorp.com/',
+		},
+		{
+			title: 'Support',
+			href: 'https://www.hashicorp.com/customer-success',
+		},
+	],
+}
+
+/**
+ * We expected posted data to be an OpenAPI spec in JSON format.
+ * We also allow an optional schema `info.description`, which normally would be
+ * included in the spec content, so that authors can more easily develop
+ * and preview their `info.description` content using our preview tool.
+ */
+type ExpectedBody = {
+	openApiJsonString: string
+	openApiDescription: string
+}
+
+export default async function handler(
```

Comment:
This is kind of the "driver" of the view - accepts OpenAPI spec `JSON` from the client, and sends back the "static props" needed to render an `OpenApiDocsView`.

Note that we've set up `GENERIC_PAGE_CONFIG` here to keep the preview input interface simpler. We _could_ in theory expose more inputs, but it didn't feel worth it at this time - we want this preview tool to focus in on the contents of the spec `.json` file.